src/core/components/accordion/index.tsx

```diff
@@ -1,27 +1,105 @@
-import React, { ReactNode } from "react"
-import { accordion, label } from "./styles"
+import React, { useState, ReactNode } from "react"
+import {
+	accordion,
+	labelText,
+	titleRow,
+	accordionRow,
+	accordionToggle,
+	showAccordionElement,
+	hideAccordionElement,
+	grey,
+	white,
+} from "./styles"
 import { Props } from "@guardian/src-helpers"
+import { SvgChevronDownSingle, SvgChevronUpSingle } from "@guardian/src-svgs"
 
-interface AccordionProps extends Props {
-	children: ReactNode
-}
+type Colours = "grey" | "white"
 
-const Accordion = ({ children }: AccordionProps) => {
-	return <div css={accordion}>{children}</div>
+const backgroundColours = {
+	grey: grey,
+	white: white,
 }
 
-interface AccordionRowProps extends Props {
+interface AccordionProps extends Props {
 	label: string
 	children: ReactNode
+	backgroundColour: Colours
 }
 
-const AccordionRow = ({ label: labelText, children }: AccordionRowProps) => {
+const Accordion = ({ children, label, backgroundColour }: AccordionProps) => {
+	const [accordionOpen, toggleAccordion] = useState(false)
+
 	return (
-		<div>
-			<strong css={label}>{labelText}</strong>
-			<div>{children}</div>
+		<div
+			css={[accordion, backgroundColours[backgroundColour]]}
+			aria-label="tablist"
+		>
+			<div css={titleRow}>
+				<strong css={labelText}>{label}</strong>
+				<button
+					aria-controls="accordionBody"
+					aria-expanded={accordionOpen}
+					css={accordionToggle}
+				>
+					<div
+						css={
+							accordionOpen
+								? showAccordionElement
+								: hideAccordionElement
+						}
+						onClick={() => toggleAccordion(false)}
+					>
+						Hide
+					</div>
+					<div
+						css={
+							accordionOpen
+								? showAccordionElement
+								: hideAccordionElement
+						}
+					>
+						<SvgChevronUpSingle />
+					</div>
```

Comment:
Maybe we can combine the label and SVG into a single div?

```jsx
<div css={...}>
  Hide
  <SvgChevronUpSingle />
</div>

```

Reply:
Yeees... this is undoubtedly my poor css chops, but I am struggling to get these to behave the way I want without quite a few divs to manage size and vertical alignment.